cmd/osbuild-base-test-runner/main.go

```diff
@@ -0,0 +1,193 @@
+// +build integration
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/osbuild/osbuild-composer/internal/boot"
+)
+
+type timeoutError struct{}
+
+func (*timeoutError) Error() string { return "" }
+
+func runLocalCommand(name string, args ...string) {
+	cmd := exec.Command(name, args...)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	if err := cmd.Run(); err != nil {
+		fmt.Println("Error: ", err)
+	}
+}
+
+func runSSHCommand(address, privateKey, command string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	cmdName := "ssh"
+	cmdArgs := []string{
+		"-p", "22",
+		"-i", privateKey,
+		"-o", "StrictHostKeyChecking=no",
+		"-o", "UserKnownHostsFile=/dev/null",
+		"redhat@" + address,
+		command,
+	}
+
+	var cmd *exec.Cmd = exec.CommandContext(ctx, cmdName, cmdArgs...)
+
+	output, err := cmd.Output()
+
+	if ctx.Err() == context.DeadlineExceeded {
+		return &timeoutError{}
+	}
+
+	if err != nil {
+		if exitError, ok := err.(*exec.ExitError); ok {
+			if exitError.ExitCode() == 255 {
+				return &timeoutError{}
+			}
+		} else {
+			return fmt.Errorf("ssh command failed from unknown reason: %#v", err)
+		}
+	}
+	outputString := strings.TrimSpace(string(output))
```

Comment:
Well, first I get the `output` and `err` values, then I check for errors and if there are none I process the output. Can you think of a better place?

Reply:
I don't understand why you trim at all? The caller of this function specifies the bianry to execute, and they know what output to expect. They should be the ones to decide whether to trim or not, right? Trimming in the helper just prevents callers from getting the raw output, in case they ever need so.

I like trimming data at the latest possible stage, if needed at all. So that would mean when you compare the result. Otherwise, we end up trimming at all stages needlessly, or even harmfully dropping content.